src/org/javaday/java8/misused/optional/HundredAndOneApproach.java

```diff
@@ -0,0 +1,114 @@
+package org.javaday.java8.misused.optional;
+
+import org.javaday.java8.misused.Annotations.Bad;
+import org.javaday.java8.misused.Annotations.Good;
+import org.javaday.java8.misused.Annotations.Ugly;
+
+import java.util.Optional;
+
+import static java.util.Optional.empty;
+import static java.util.Optional.ofNullable;
+
+public class HundredAndOneApproach {
+
+    @Ugly
+    class SameOldImparativeStyle {
+
+        String getPersonCarInsuranceName(Person person) {
+            String name = "Unknown";
+            if (ofNullable(person).isPresent()) {
+                if (person.getCar().isPresent()) {
+                    if (person.getCar().get().getInsurance().isPresent()) {
+                        name = person.getCar().get().getInsurance().get().getName();
+                    }
+                }
+            }
+            return name;
+        }
+    }
+
+    @Ugly
+    class UsingIfPresentInSameImparativeWayWithDirtyHack{
+
+        String getPersonCarInsuranceName(Person person) {
+            final StringBuilder builder = new StringBuilder();
+            ofNullable(person).ifPresent(
+                    p -> p.getCar().ifPresent(
+                            c -> c.getInsurance().ifPresent(
+                                    i -> builder.append(i.getName())
+                            )
+                    )
+            );
+            return builder.toString();
+        }
+    }
+
+    @Bad
+    class UsingMapWithUncheckedGet {
+
+        String getPersonCarInsuranceName(Person person) {
+            return ofNullable(person)
+                    .map(Person::getCar)
+                    .map(car -> car.get().getInsurance())
+                    .map(insurance -> insurance.get().getName())
+                    .orElse("Unknown");
+        }
+    }
+
+    @Ugly
+    class KeepItSimple {
```

Comment:
What is bad we will spend too more time on this sample, especially taking into account every sample requires explanation.